File: exercism/javascript/hamming/hamming.js
function Hamming() {};

Hamming.prototype.compute = function(a,b) {
  // make sure strands are same length
  if (a.length != b.length) { throw "DNA strands must be of equal length." }
  // counter for missing matches
  var count = 0;

  for (var i = 0; i < a.length; i++) {
    a[i] != b[i] ? count++ : count
  }

  return count
}

module.exports = Hamming;

File: exercism/ruby/hamming/hamming.rb
class Hamming
  def self.compute(a, b)
    raise ArgumentError, 'Must be same length' unless a.length == b.length

    diff = 0
    a.each_char.with_index { |v, i| v != b[i] ? diff += 1 : diff}
    diff
  end
end

module BookKeeping
  VERSION = 3
end
